File: data/src/main/java/com/github/gmetal/data/entity/BaseMoviesDbEntity.java
package com.github.gmetal.data.entity;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

@JsonObject
public class BaseMoviesDbEntity<T> {
    // This is parameterised so that we can get the code to compile :)

    @JsonField(name = "status_message")
    public String statusMessage;

    @JsonField(name = "status_code")
    public int statusCode;
}

File: data/src/main/java/com/github/gmetal/data/entity/BaseMoviesDbEntity.kt
package com.github.gmetal.data.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable


@Serializable
open class BaseMoviesDbEntity {
    @SerialName("status_message")
    var statusMessage: String? = null

    @SerialName("status_code")
    var statusCode: Int = 0
}
